cloudify_nsx/network/esg.py

```diff
@@ -69,11 +69,12 @@ def create(**kwargs):
     use_existing, edge_dict = common.get_properties_and_validate(
         'edge', kwargs, validation_rules
     )
+    edge_dict = common.possibly_assign_vm_creation_props(edge_dict)
 
     resource_id = ctx.instance.runtime_properties.get('resource_id')
 
     # credentials
-    client_session = common.nsx_login(kwargs)
+    client_session = common.nsx_login(edge_dict)
```

Comment:
Looks little strange, in common.nsx_login we want to use 'nsx_auth', so in current case we need kwargs for case when we want overwrite nsx_auth from operation inputs. (it don't make big sense, but for case when user want to get auth from attributes(not properties). )